api/src/main/java/javax/mvc/engine/ViewEngineContext.java

```diff
@@ -74,21 +71,6 @@
     Models getModels();
 
     /**
-     * Returns HTTP request object from the Servlet container.
-     *
-     * @return HTTP request object.
-     */
-    HttpServletRequest getRequest();
-
-    /**
-     * Returns HTTP response object from the servlet container. The underlying
-     * output stream should be used to write the result of processing a view.
-     *
-     * @return HTTP response object.
-     */
-    HttpServletResponse getResponse();
-
```

Comment:
> Think I once was in a situation where someone set a caching header before me and the only thing I could do is setting the header again with a nonsense value. But can't event tell you the used framework any more...

So you say having a way to delete headers would be required? Maybe this could be useful. But I still not like providing a mutable map to the user. That feels wrong. We could also add `ViewEngineContext#removeHeader(String)`.

But the next problem is that a header can be set multiple times and therefore have multiple values, right? I think that's the reason JAX-RS uses `MultivaluedMap`. :)

> Didn't try but I think Object could be List if the same header is set twice.

The map is already a "multivalued map". So every key has multiple values. So I don't think that's the reason they chose `Object`.

> Without adding the PrintWriter to the interface we don't have any chance to control the encoding from outside the ViewEngine. I'm not too happy with such a decision because I would prefer @Produces. [...]

Hmmm. Note sure if adding such a global property isn't way more confusing. Then we have three places where we have an encoding specified: `@Produces`, the config property and within the template (like in the JSP case).

What about something like this:

  * JAX-RS already uses `@Produces` for the matching algorithm which finds the corresponding controller method. That's fine. Users *may* set an encoding here with this annotation if they want to.
  * Why not passing this `MediaType` (specified via `@Produces`) to the view engine? That should be easy, because `MessageBodyWriter` gets the `MediaType` as a parameter.
  * The view engine now knows about the media type from the `@Produces` annotation. So we could specify that the view engine must use the encoding specified in the media type. 
  * If the view engine finds some other encoding specified in the template (like in the JSP) page, we could specify that the view engine must prefer this one. If this also generates some warning log message in the logs is up to the view engine.

Doesn't this make sense? We would have the following cases:

  * No `@Produces`, no encoding in template: -> UTF8  (@lefloh: correct?)
  *  No `@Produces`, encoding ISO-8859-1 in the template -> ISO-8859-1
  * `@Produces` with ISO-8859-1, no encoding in template -> ISO-8859-1
  * `@Produces` with ISO-8859-1, encoding UTF-8 in the template -> UTF8 + warning in the logs 

Thoughts?